lib/service/storage/elasticsearch.js

```diff
@@ -822,6 +822,77 @@ class ElasticSearch extends Service {
     }
   }
 
+  /**
+   * Delete fields of a document and replace it
+   *
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {String} id - Document id
+   * @param {Array}  fields - Document fields to be removed
+   * @param {Object} options - refresh (undefined), userId (null)
+   *
+   * @returns {Promise.<{ _id, _version, _source }>}
+   */
+  async deleteFields (
+    index,
+    collection,
+    id,
+    fields,
+    { refresh, userId=null } = {})
+  {
+    const esIndex = this._getESIndex(index, collection);
+    const esRequest = {
+      id,
+      index: esIndex,
+    };
+
+    // metadata
+    const _kuzzle_info = {
+      author: getUserId(userId),
+      updatedAt: Date.now(),
+      updater: getUserId(userId)
+    };
+
+    try {
+      const { body: exists } = await this._client.exists({ id, index: esIndex });
+
+      if (!exists) {
+        throw kerror.get('not_found', id, index, collection);
+      }
+
+      debug('DeleteFields document: %o', esRequest);
+      const { body } = await this._client.get(esRequest);
+
+      for (const field of fields) {
+        _.set(body._source, field, undefined);
+      }
+
+      body._source._kuzzle_info = Object.assign(
+        body._source._kuzzle_info || { createdAt: Date.now() },
+        _kuzzle_info);
+
+      const newEsRequest = {
+        body: body._source,
+        id,
+        index: esIndex,
+        refresh
+      };
+
+      assertNoRouting(newEsRequest);
+      assertWellFormedRefresh(newEsRequest);
+
+      await this._client.index(newEsRequest);
+
+      return {
+        _id: id,
+        _source: body._source,
+        _version: body._version
+      };
```

Comment:
The version you return in your response is the one of the document you got from ES, before the fields are deleted. You need to get the new version from ES' response:

```suggestion
      const { body: updated } = await this._client.index(newEsRequest);

      return {
        _id: id,
        _source: updated._source,
        _version: updated._version
      };
```